Added functionality to get all unavailable products

JethroWarnett opened the pull request:
A seller can query the database for all of his products, that are approved and unavailable

JethroWarnett committed:
Added functionality to get all unavailable products

A seller can query the database for all of his products, that are approved and unavailable

dominiqueroux commented on backend/src/services/product.service.ts:
```diff
@@ -3,6 +3,8 @@ import { User } from '../models/user.model';
 import { ProductImageAttributes, ProductImage } from '../models/productImage.model';
 import { upload, MulterRequest } from '../middlewares/imageUpload';
+import {Op} from "sequelize";
+import {userInfo} from "os";
```
Is this really needed?

JethroWarnett replied:
Op yes, 
userInfo I didn't even notice. Seems to be outdated code, that wasn't deleted.

JethroWarnett committed:
Added the missing unit tests

Added the unit test for getUnavailableProducts and get availableProducts. Also removed the unnecessary import in product.service.ts.

JethroWarnett committed:
Revert "Added the missing unit tests"

This reverts commit e9c4fe399e6529713e9bcc4c1b1ee8f6160e746a.

JethroWarnett committed:
Added missing unit tests

Fixed the import and added the unit tests for getAvailableProducts and getUnavailableProducts
